### DISCUSSION TITLE
Allow conditions fields to be dynamic

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#7123] [Archive]
Hi.

`Fields Conditions` is a great feature. I was wondering if we can make it dynamic so that fields hide/show when we change the value of one fields right away when we add or edit items. Like if I have a dropdown with values and depending on that filds needs to be hide/show, would be really great. Do you think is possible?

**Comment by rijkvanzanten** [comment]
This should already be possible!

![CleanShot 2021-08-02 at 09 53 58@2x](https://user-images.githubusercontent.com/9141017/127872855-35806431-ed29-4067-bad6-0e73d29d0bfb.png)

**Reply by botzill** [reply]
Yes, it does apply, but only if we save the item and open it again. What if we add new item, and in that form while we select items from the dropdown or O2M relation to hide/show fields?

**Reply by rijkvanzanten** [reply]
> Yes, it does apply, but only if we save the item and open it again

Ahh, that might be because the difference of updated item vs existing PK, can you try setting the rule to:

```json
{
  "_or": [
    {
      "my-field": {
        "_eq": "my-value"
      }
    },
    {
      "my-field": {
        "id": {
          "_eq": "my-value"
        }
      }
    }
  ]
}
```

(make sure to substitute `my-field`, `id`, and `my-value` for the field name, primary key, and actual value you're looking for respectively

**Reply by botzill** [reply]
I did try this:

```json
{
    "_or": [
        {
            "country": {
                "name": {
                    "_nin": [
                        "Algeria"
                    ]
                }
            }
        },
        {
            "country": {
                "_ne": "Algeria"
            }
        }
    ]
}
```

I have country as:

<img width="837" alt="Screen Shot 2021-08-02 at 19 40 31" src="https://user-images.githubusercontent.com/4686040/127895449-b7b38452-cd84-425c-b8a1-b35eac30f5f4.png">

<img width="880" alt="Screen Shot 2021-08-02 at 19 40 47" src="https://user-images.githubusercontent.com/4686040/127895483-b70c8387-aa9e-4f10-a875-6eb40076bb19.png">

If country is `Algeria` I want to show some fields, but they are not showing:

<img width="830" alt="Screen Shot 2021-08-02 at 19 42 26" src="https://user-images.githubusercontent.com/4686040/127895631-70f5726d-357f-41cf-995b-c8464191bb7f.png">

**Reply by rijkvanzanten** [reply]
I think you might have to invert the operation here (eg make it hidden by default, and then un-hide it in the rule), as the current "or" statement means that one of those two will always match, which will set the field to hidden 🤔

**Reply by botzill** [reply]
So, if I make the field hidden by default and in the rule I do not set `Hidden` it will un-hide, right?

**Reply by botzill** [reply]
I tried to reverse, but does not apply, it always show the fields:

<img width="829" alt="Screen Shot 2021-08-02 at 19 54 33" src="https://user-images.githubusercontent.com/4686040/127896996-d23a452d-c447-4894-a761-501c43111113.png">

I set the field default to:
<img width="610" alt="Screen Shot 2021-08-02 at 19 55 48" src="https://user-images.githubusercontent.com/4686040/127897159-9cd887aa-14e7-47b0-876d-386589069af6.png">

**Reply by rijkvanzanten** [reply]
Odd, that should do the trick. lemme do some testing 👍🏻

**Reply by botzill** [reply]
Thx a lot!

**Reply by rijkvanzanten** [reply]
Found a small issue! That nested field rule was always seen as valid. https://github.com/directus/directus/pull/7135 fixed that 👍🏻

**Reply by botzill** [reply]
After upgrade, I see the same results. I test with hiding by default and the condition should make it visible, but is still hidden. Any ideas?

**Reply by rijkvanzanten** [reply]
This seems to work for me as expected now. I've added an `author` many to one field, and an `author_note` field that should only be displayed when `author` is set to `1`. `author_note`'s `hidden` is set to `true`, `conditions` is set to 

```json
[
  {
    "name": "When author is 1",
    "rule": {
      "_or": [
        {
          "author": {
            "_eq": 1
          }
        },
        {
          "author": {
            "id": {
              "_eq": 1
            }
          }
        }
      ]
    },
    "hidden": false
  }
]
```

**Reply by botzill** [reply]
Hm, can you test with another field then `id` for M2O relation? I did a test with `id` and seems to work.

**Reply by botzill** [reply]
Plus, when we open the form to add a new item and no selection is made for `m2o` relation is does not apply the rule?

**Reply by rijkvanzanten** [reply]
> Hm, can you test with another field then id for M2O relation? I did a test with id and seems to work.

That would make sense, as the value for that field may or may not exist. If the user doesn't alter that specific field, it's value won't be staged and thus don't exist to be checked against 🤔

**Reply by botzill** [reply]
Hm, So, it will work only with `id` ?

**Reply by rijkvanzanten** [reply]
@botzill Yes, unless the user has explicitly changed the value of the field

**Comment by botzill** [comment]


**Reply by rijkvanzanten** [reply]
Within the next 30 or so minutes 👍🏻

**Comment by Jeyakumarrathnaselvi** [comment]
Any outcome here?
